crates/pallet-erc20-support/src/migrations/v1.rs

```diff
@@ -0,0 +1,53 @@
+//! Migration to Version 1.
+
+use frame_support::{dispatch::GetStorageVersion, sp_tracing::info, traits::Get, weights::Weight};
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+use crate::BalanceOf;
+use crate::{Approvals, Config, Pallet};
+
+/// Migrate from version 0 to 1.
+pub fn migrate<T: Config<I>, I: 'static>() -> Weight {
+    let current = <Pallet<T, I>>::current_storage_version();
+    let onchain = <Pallet<T, I>>::on_chain_storage_version();
+
+    // Read the onchain version.
+    let mut weight: Weight = T::DbWeight::get().reads(1);
+
+    info!(message = "Running migration to v1", from = ?onchain);
+
+    if onchain == 1 {
+        info!(message = "Already at version 1, nothing to do");
+        return weight;
+    }
```

Comment:
What is `onchain` is 2? I think this migration is only safe to run when the `onchain` version is `0`, not when it's not `1`.

Reply:
There is no `onchain` 2. The migration from `0` to `1`.

Reply:
Redesigned based on discussion - https://github.com/humanode-network/humanode/pull/809/commits/e87a191a4722ca14acf015006cb8044b456b50a6